src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,573 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Mathieu Carrière
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import numpy as np
+import itertools
+import matplotlib
+import matplotlib.pyplot as plt
+
+from sklearn.base            import BaseEstimator, TransformerMixin
+from sklearn.cluster         import DBSCAN, AgglomerativeClustering
+from sklearn.metrics         import pairwise_distances
+from scipy.spatial.distance  import directed_hausdorff
+
+from .. import SimplexTree, CoverComplex
+
+class CoverComplexPy(BaseEstimator, TransformerMixin):
+    """
+    This is a mother class for MapperComplex, GraphInducedComplex and NerveComplex.
+
+    Attributes:
+        simplex_tree (gudhi SimplexTree): simplicial complex representing the cover complex computed after calling the fit() method.
```

Comment:
Scikit-learn says that all estimated attributes should have a name that ends in `_` (https://scikit-learn.org/stable/developers/develop.html#estimated-attributes).

Reply:
Good catch. Added an `_` for `simplex_tree` and `node_info`.